hera_mc/weather.py

```diff
@@ -0,0 +1,128 @@
+    variable: name of weather variable. One of the keys in weather_sensor_dict (String, part of primary_key)
+    value: measured value (Float)
+    units: units of value (String)
+    """
+    __tablename__ = 'weather_data'
+    time = Column(BigInteger, primary_key=True)
+    variable = Column(String, nullable=False, primary_key=True)
+    value = Column(Float, nullable=False)
+    units = Column(String, nullable=False)
+
+    @classmethod
+    def create(cls, time, variable, value):
+        """
+        Create a new weather object.
+
+        Parameters:
+        ------------
+        time: astropy time object
+            astropy time object based on a timestamp from the katportal sensor.
+        variable: string
+            must be a key in weather_sensor_dict
+        value: float
+            value from the sensor associated with the variable
+        """
+        if not isinstance(time, Time):
+            raise ValueError('time must be an astropy Time object')
+        if variable not in weather_sensor_dict.keys():
+            raise ValueError('variable must be a key in weather_sensor_dict.')
+        if not isinstance(value, (float)):
+            raise ValueError('value must be a float.')
+        weather_time = floor(time.gps)
+
+        return cls(time=weather_time, variable=variable, value=value,
+                   units=weather_sensor_dict[variable]['units'])
+
+
+@tornado.gen.coroutine
+def create_from_sensors(starttime, stoptime, variables=None):
+    """
+    Create a list of weather objects from sensor data.
+
+    Parameters:
+    ------------
+    starttime: astropy time object
+        time to start getting history.
+    stoptime: astropy time object
+        time to stop getting history.
+    variable: string
+        variable to get history for. Must be a key in weather_sensor_dict,
+        defaults to all keys in weather_sensor_dict
+
+    Returns:
+    -----------
+    A list of WeatherData objects
+    """
+    from katportalclient import KATPortalClient
+
+    if not isinstance(starttime, Time):
+        raise ValueError('starttime must be an astropy Time object')
+
+    if not isinstance(stoptime, Time):
+        raise ValueError('stoptime must be an astropy Time object')
+
+    if variables is None:
+        variables = weather_sensor_dict.keys()
+    elif not isinstance(variables, (list, tuple)):
+        variables = [variables]
+
+    sensor_names = []
+    sensor_var_dict = {}
+    for var in variables:
+        if var not in weather_sensor_dict.keys():
+            raise ValueError('variable must be a key in weather_sensor_dict')
+        sensor_names.append(weather_sensor_dict[var]['sensor_name'])
+        sensor_var_dict[weather_sensor_dict[var]['sensor_name']] = var
+
+    portal_client = KATPortalClient('http://portal.mkat.karoo.kat.ac.za/api/client',
+                                    on_update_callback=None)
+
+    sensor_names = yield portal_client.sensor_names(sensor_names)
```

Comment:
So here it seems that we're filtering down our list of sensors to use; the docs to the `sensor_names` method say that the inputs are regular expressions. But we're not using any regexes, so I think at best all this call does is trigger an error if a sensor is unknown. But that will happen later if that's the case. Given how our code works, I think it would be better just to delete this line.

Reply:
ok